HandheldCompanion/Devices/MSI/ClawA1M.cs

```diff
@@ -0,0 +1,179 @@
+using System.Numerics;
+using System.Threading;
+using System.Threading.Tasks;
+using WindowsInput.Events;
+
+namespace HandheldCompanion.Devices;
+
+public class ClawA1M : IDevice
+{
+    private enum WMIEventCode
+    {
+        LaunchMcxMainUI = 41, // 0x00000029
+        LaunchMcxOSD = 88, // 0x00000058
+    }
+
+    private readonly Dictionary<WMIEventCode, ButtonFlags> keyMapping = new()
+    {
+        { 0, ButtonFlags.None },
+        { WMIEventCode.LaunchMcxMainUI, ButtonFlags.OEM1 },
+        { WMIEventCode.LaunchMcxOSD, ButtonFlags.OEM2 },
+    };
+
+    private ManagementEventWatcher specialKeyWatcher;
+    private ManagementScope scope;
+
+    public ClawA1M()
+    {
+        // device specific settings
+        ProductIllustration = "device_msi_claw";
+
+        // https://www.intel.com/content/www/us/en/products/sku/236847/intel-core-ultra-7-processor-155h-24m-cache-up-to-4-80-ghz/specifications.html
+        nTDP = new double[] { 28, 28, 65 };
+        cTDP = new double[] { 20, 65 };
+        GfxClock = new double[] { 100, 2250 };
+        CpuClock = 4800;
+
+        GyrometerAxis = new Vector3(1.0f, -1.0f, -1.0f);
+        GyrometerAxisSwap = new SortedDictionary<char, char>
+        {
+            { 'X', 'X' },
+            { 'Y', 'Z' },
+            { 'Z', 'Y' }
+        };
+
+        AccelerometerAxis = new Vector3(1.0f, 1.0f, -1.0f);
+        AccelerometerAxisSwap = new SortedDictionary<char, char>
+        {
+            { 'X', 'X' },
+            { 'Y', 'Z' },
+            { 'Z', 'Y' }
+        };
+
+        // device specific capacities
+        Capabilities |= DeviceCapabilities.None;
+
+        OEMChords.Add(new DeviceChord("CLAW",
+            new List<KeyCode>(), new List<KeyCode>(),
+            false, ButtonFlags.OEM1
+        ));
+
+        OEMChords.Add(new DeviceChord("QS",
+            new List<KeyCode>(), new List<KeyCode>(),
+            false, ButtonFlags.OEM2
+        ));
+
+        OEMChords.Add(new DeviceChord("M1",             // unimplemented
+            new List<KeyCode>(), new List<KeyCode>(),
+            false, ButtonFlags.OEM3
+        ));
+
+        OEMChords.Add(new DeviceChord("M2",             // unimplemented
+            new List<KeyCode>(), new List<KeyCode>(),
+            false, ButtonFlags.OEM4
+        ));
+
+        // start thread to monitor WMI events
+        new Thread(() =>
+        {
+            try
+            {
+                scope = new ManagementScope("\\\\.\\root\\WMI");
+                scope.Connect();
+                if (!scope.IsConnected)
+                    return;
+                specialKeyWatcher = new ManagementEventWatcher(scope, (EventQuery)(new WqlEventQuery("SELECT * FROM MSI_Event")));
+                specialKeyWatcher.EventArrived += onWMIEvent;
+            }
+            catch (Exception ex)
+            {
+                LogManager.LogInformation("Exception configuring MSI_Event monitor: {0}", ex.Message);
+            }
+        }).Start();

```

Comment:
I've removed scope as a private field in this class.  I've removed the thread logic to simplify the class.  I moved the WMI event setup and tear down to their own methods with exception catching and logging.  I marked specialKeyWatcher as nullable since it is not initialized in the constructor anymore.